### PR TITLE
updates tutorial about how to use enviroment

### PR DESCRIPTION
**Pull request by laerus**
#1004

### CODE REVIEW

File: docs/tutorial.rst [hunk 1]
```diff
@@ -136,19 +136,54 @@ variable in Python.  The same is true for deleting them too.
 
 Very nice.
 
-__xonsh_env__
+Environment Lookup with ``${}``
```
Comment (scopatz) [inline docs/tutorial.rst]: I think this section, should still follow the section it now comes after.  The reader should know about the environment before they learn about expression based lookup.

**Comment by scopatz** [comment]
Thanks for putting this in @laerus!

File: docs/tutorial.rst (hunk 1 above)
Comment (laerus) [inline docs/tutorial.rst]: i liked that flow, you talk about envvars with `$NAME` then you introduce something better and the last example leads to ellipsis which then you exlain. but np! i'll just reverse this two sections.

File: docs/tutorial.rst [hunk 2]
```diff
@@ -136,6 +136,27 @@ variable in Python.  The same is true for deleting them too.
 
 Very nice.
 
+The Environment ``${...}``
+--------------
```
Comment (scopatz) [inline docs/tutorial.rst]: This should be `===` and needs to extend below the whole line.

File: docs/tutorial.rst [hunk 3]
```diff
@@ -136,6 +136,27 @@ variable in Python.  The same is true for deleting them too.
...
+
+All environment variables live in the built-in ``${...}`` (aka ``__xonsh_env__``) mapping.
+You can access this mapping directly, but in most situations, you shouldn’t need to.
+
+One helpful method on the ``${...}`` is :func:`~xonsh.environ.Env.swap`.  It can be used to temporarily set an
```
Comment (scopatz) [inline docs/tutorial.rst]: Line too long.

File: docs/tutorial.rst (hunk 2 above)
Comment (scopatz) [inline docs/tutorial.rst]: Nevermind, it should be `---`, but it still needs to extend the whole way

File: docs/tutorial.rst [hunk 4]
```diff
@@ -136,6 +136,27 @@ variable in Python.  The same is true for deleting them too.
 
 Very nice.
 
+The Environment ``${...}``
```
Comment (scopatz) [inline docs/tutorial.rst]: Maybe this could be `The Environment Itself ${...}`

File: docs/tutorial.rst [hunk 5]
```diff
@@ -157,6 +157,40 @@ environment variable:
 
     >>>
 
+Environment Lookup with ``${}``
+--------------
```
Comment (scopatz) [inline docs/tutorial.rst]: Line needs to extend the whole way.

File: docs/tutorial.rst [hunk 6]
```diff
@@ -157,6 +157,40 @@ environment variable:
 
     >>>
 
+Environment Lookup with ``${}``
```
Comment (scopatz) [inline docs/tutorial.rst]: Maybe set this to `${<expr>}` to make it clearly distinct

File: docs/tutorial.rst [hunk 7]
```diff
@@ -157,6 +157,40 @@ environment variable:
...
+Not bad, xonsh, not bad.
+
+If you want to check if an environment variable is present in your current
+session (say, in your awesome new ``xonsh`` script) you can pass an Ellipsis to
+the ``${}`` operator:
+
+.. code-block:: xonshcon
+
+   >>> 'HOME' in ${...}
+   True
```
Comment (scopatz) [inline docs/tutorial.rst]: I would move this paragraph to the above section now :)

**Comment by scopatz** [comment]
Thanks for dealing with all of my edits :)

**Comment by laerus** [comment]
poor travis

**Comment by scopatz** [comment]
> poor travis

You can always cancel jobs :)

**Comment by laerus** [comment]
should look that up

File: docs/tutorial.rst [hunk 8]
```diff
@@ -157,6 +163,31 @@ environment variable:
 
     >>>
 
+Environment Lookup with ``${<expr>}``
+-------------------------------
```
Comment (scopatz) [inline docs/tutorial.rst]: This line need to extend the whole width of the line above.

**Comment by scopatz** [comment]
Thanks @laerus - just the one more small issue.

**Comment by laerus** [comment]
thank you for your patience!

**Comment by scopatz** [comment]
No problem! My pleasure.

**Pull request merged**